malcolm/modules/scanning/parts/directorymonitorpart.py

```diff
@@ -0,0 +1,42 @@
+from annotypes import Anno, add_call_types
+
+from malcolm.core import PartRegistrar
+from malcolm.modules.builtin.hooks import AContext
+from malcolm.modules.builtin.parts import AMri, APartName, ChildPart
+
+from .. import hooks
+
+with Anno("Whether to raise a ValueError for a bad status"):
+    AErrorOnFail = bool
+
+
+class DirectoryMonitorPart(ChildPart):
+    """Part for checking a directoryMonitor Manager is happy"""
+
+    def __init__(
+        self, name: APartName, mri: AMri, error_on_fail: AErrorOnFail = True
+    ) -> None:
+        super().__init__(name, mri, initial_visibility=True)
+        self.error_on_fail = error_on_fail
+
+    def setup(self, registrar: PartRegistrar) -> None:
+        super().setup(registrar)
+        # Hooks
+        registrar.hook(hooks.ConfigureHook, self.check_directories)
+
+    @add_call_types
+    def check_directories(self, context: AContext) -> None:
+        child = context.block_view(self.mri)
+        # Perform check
+        getattr(child, "managerCheck")()
```

Comment:
```suggestion
        child.managerCheck()
```

Reply:
Does this support caput callback? There might still be a race condition here between getting the callback and reading the monitored managerStatus value. You might want to put in a 0.1s sleep (ugg) to guard against this

Reply:
It does, so the status value should be updated before it gets checked (assuming that just calling managerCheck() will wait for this callback, or does Malcolm not wait for the callback by default?).

Reply:
Looks like it does wait by default in the CAActionPart, and I didn't encounter any problems in testing (I changed the file mounts to a bad/good state and these changes were picked up by this part on configure - without any additional polling in place)

Reply:
Malcolm monitors all PVs. There is a race condition between the caput callback returning and the monitored PV updating. This is a CA limitation. You can only get around this by cagetting the PV rather than monitoring it. 

Actually, this is exactly what the CAActionPart does. If you set the `status_pv` of managerCheck to whatever managerStatus points to, then it will do this check for you and raise an AssertionError if `status_pv` doesn't match

Reply:
I see, the monitoring is the problem. I have now refactored to use the status_pv and checking it against the good value.